### DISCUSSION TITLE
Can I define the assets type?

### DISCUSSION THREAD

**Discussion by Benbinbin** [vuepress/core#253] [Q&A] [answered]
I want to link a `.pdf` file in Markdown with relative paths in project, when I click the link in development environment it can show the pdf file, but it can't show the file correctly when I build the project, it show `Cannot GET /some-path-of-the-markdown-file/xxx.pdf` error.

I use the default theme and Vite as a bundler, refer to the vite documentation it support to [extend the assets type](https://vitejs.dev/guide/assets.html#:~:text=You%20can%20extend%20the%20internal%20list%20using%20the%20assetsInclude%20option.) by using the [assetsInclude](https://vitejs.dev/config/#assetsinclude) option. I set this option in `.vuepress/config.js` but it doesn't work

```js
module.exports = {
  // ...
  bundlerConfig: {
    viteOptions: {
      assetsInclude: /\.pdf$/
    }
}
```

when I use webpack as the bundler in VuePress v1, it can use `file-loader` to define the asset types.

**Comment by Mister-Hope** [comment]
Just to be curious, what's the benifit to regard pdf as an asset? It surely has disadvantages such as increasing build and dev memory usage and a little more build time.

You could just place it in the public folder and use an absolute link.

VuePress supports images and code blocks only because they can be injected and displayed in page, but pdf can't.

**Reply by Benbinbin** [reply]
I take many notes when learning skill in Markdown files, which have a lot of other type of files links, and I want to use VuePress to compile these Markdown files to html,  so I can edit them locally and read them online.

**Comment by Mister-Hope** [comment, marked as answer]
If you do want this feature, a PR without slowing down current speed is welcomed. We may not have plans supporting this feature as we would regard this as an edge case.

You do have workarounds like using absolute links or creating a markdown-it plugin for that. The built in code import plugin may help you to achieve that in markdown-it.

Resources like PDFs really should not be regarded as assets.
